devito/archinfo.py

```diff
@@ -64,12 +66,16 @@ def get_cpu_brand():
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
+        if cpu_info['brand'] is None:
+            cpu_info['brand'] = cpuinfo.get_cpu_info().get('raw_arch_string')
 
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    try:
+        if not ('flags' in cpu_info):
```

Comment:
`if 'flags' not in cpu_info`

Reply:
actually, this should be moved [here](https://github.com/devitocodes/devito/blob/master/devito/archinfo.py#L46) in place of the `return None`. Right?

Reply:
if we do so, `cpuinfo.get_cpu_info()` will be called twice. Then we probably want to memoize the call.

```
@memoized_func
def _get_cpu_info_from_cpuinfo():
     return cpuinfo.get_cpu_info()
```

unless it's already memoized internally by cpuinfo itself. Can you find out and/or time it

Reply:
we already have:
https://github.com/devitocodes/devito/pull/1515/files#diff-fc43288ece59fd09fd1aaeaddf92cd861a8289feb72e30d449719f1dae337f3bR20
```
@memoized_func
def get_cpu_info():

```
I guess it is enough?